service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -234,8 +235,12 @@ private String getSupervisionOrderExpiryDate(OrderTypeAndDocument typeAndDocumen
     }
 
     private String getInterimExpiryDate(InterimEndDate interimEndDate) {
-        return interimEndDate.toLocalDateTime()
-            .map(dateTime -> formatLocalDateTimeBaseUsingFormat(dateTime, TIME_DATE))
-            .orElse(END_OF_PROCEEDINGS.getLabel());
+        if (interimEndDate.getType().equals(SPECIFIC_TIME_NAMED_DATE)) {
+            return formatLocalDateTimeBaseUsingFormat(interimEndDate.getEndDateTime(), TIME_DATE);
+        } else {
+            return interimEndDate.toLocalDateTime()
+                .map(dateTime -> formatLocalDateTimeBaseUsingFormat(dateTime, TIME_DATE))
+                .orElse(END_OF_PROCEEDINGS.getLabel());
+        }
```

Comment:
Thoughts on making use of the optional nature of the second option like so?

```suggestion
        return Optional.ofNullable(interimEndDate.getEndDateTime())
            .or(interimEndDate::toLocalDateTime)
            .map(dateTime -> formatLocalDateTimeBaseUsingFormat(dateTime, TIME_DATE))
            .orElse(END_OF_PROCEEDINGS.getLabel());
```

Reply:
nice!! really liked this approach + tested, it works.!